pandas/core/frame.py

```diff
@@ -4105,9 +4105,8 @@ def _set_item_mgr(self, key, value: ArrayLike) -> None:
         if len(self):
             self._check_setitem_copy()
 
-    def _iset_item(self, loc: int, value) -> None:
-        arraylike = self._sanitize_column(value)
-        self._iset_item_mgr(loc, arraylike, inplace=True)
+    def _iset_item(self, loc: int, value: Series) -> None:
+        self._iset_item_mgr(loc, value._values, inplace=True)
```

Comment:
i think this is OK only because iset_item is called only from replace_columnwise which guarantees no reindex is necessary. can you add a comment to that effect?

Reply:
Added a comment, we have to keep the copy until reference tracking  for setitem works properly (meaning the value that is set into the DataFrame).